start.py

```diff
@@ -0,0 +1,121 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import yaml
+from socket import getfqdn, socket
+from time import sleep, time
+
+from clusterdock.models import Cluster, client, Node
+from clusterdock.utils import wait_for_condition
+
+logger = logging.getLogger('clusterdock.{}'.format(__name__))
+
+DEFAULT_NAMESPACE = 'clusterdock'
+MCS_SERVER_PORT = 8443
+
+def main(args):
+    image_prefix = '{}/{}/clusterdock:mapr{}'.format(args.registry,
+                                                     args.namespace or DEFAULT_NAMESPACE,
+                                                     args.mapr_version)
+    primary_node_image = '{}_{}'.format(image_prefix, 'primary-node')
+    secondary_node_image = '{}_{}'.format(image_prefix, 'secondary-node')
+
+    node_disks = yaml.load(args.node_disks)
+
+    # MapR-FS needs each fileserver node to have a disk allocated for it, so fail fast if the
+    # node disks map is missing any nodes.
+    if set(args.primary_node + args.secondary_nodes) != set(node_disks.keys()):
+        raise Exception('Not all nodes are accounted for in the --node-disks dictionary')
+
+    primary_node = Node(hostname=args.primary_node[0],
+                        group='primary',
+                        image=primary_node_image,
+                        ports=[{MCS_SERVER_PORT: MCS_SERVER_PORT}
+                               if args.predictable
+                               else MCS_SERVER_PORT],
+                        devices=node_disks.get(args.primary_node[0]))
+
+    secondary_nodes = [Node(hostname=hostname,
+                            group='secondary',
+                            image=secondary_node_image,
+                            devices=node_disks.get(hostname))
+                       for hostname in args.secondary_nodes]
+
+
+    cluster = Cluster(primary_node, *secondary_nodes)
+    cluster.primary_node = primary_node
+    cluster.start(args.network, pull_images=args.always_pull)
+
+    # Verify that service MapR warden is running and then only proceed.
+    logger.info('Check if Service MapR warden is running ...')
+    start_time = time()
```

Comment:
Might be better to use the `wait_for_condition` util function instead of creating your own?